### PR TITLE
Simplify and re-arrange the OpenShift guide

### PR DESCRIPTION
**Pull request by mkouba**
This is my first attempt to simplify and re-arrange the OpenShift guide.

The major changes are:
- added "Log Into the OpenShift Cluster" section
- merged the "Build and Deploy" sections; the "build and deployment in a single step" info is now mentioned first

### CODE REVIEW

File: docs/src/main/asciidoc/deploying-to-openshift.adoc [hunk 1]
```diff
@@ -53,27 +51,92 @@ When we added the OpenShift extension to the command line invocation above, the
...
+
+TIP: You can request the token via the _Copy Login Command_ link in the OpenShift web console.
+
+Finally, you don't need to use the OpenShift CLI at all.
+Instead, set the `quarkus.kubernetes-client.master-url` config property and authenticate with the `quarkus.kubernetes-client.token`, or `quarkus.kubernetes-client.username` and `quarkus.kubernetes-client.password` respectively.
+
+.Log In `application.properties` Example
+[source,properties]
+----
+quarkus.kubernetes-client.master-url=myServerUrl
```
Comment (maxandersen) [inline docs/src/main/asciidoc/deploying-to-openshift.adoc:82]: we shouldn't recommend users to put these in application properties should we ?

better do it using `mvn -Dquarkus.kubernets-client.token=... etc` ?

Comment (mkouba) [reply docs/src/main/asciidoc/deploying-to-openshift.adoc:82]: Yes, that's a good point. Let's change the example...

File: docs/src/main/asciidoc/deploying-to-openshift.adoc [hunk 2]
```diff
@@ -53,27 +51,91 @@ When we added the OpenShift extension to the command line invocation above, the
...
 [source,bash,subs=attributes+]
 ----
- ./mvnw quarkus:add-extension -Dextensions="openshift"
+./mvnw clean package -Dquarkus.kubernetes.deploy=true
 ----
- 
-Building is handled by the link:container-image#s2i[container-image-s2i] extension. To trigger a build:
+
+This command will build your application locally, then trigger a container image build and finally apply the generated OpenShift resources automatically.
+The generated resources are using OpenShift's `DeploymentConfig` that is configured to automatically trigger a redeployment when a change in the `ImageStream` is noticed.
```
Comment (geoand) [inline docs/src/main/asciidoc/deploying-to-openshift.adoc:97]:
```suggestion
The generated resources use OpenShift's `DeploymentConfig` that is configured to automatically trigger a redeployment when a change in the `ImageStream` is noticed.
```

**Review by geoand — COMMENTED** [review]
Nice improvement! I just added a tiny suggestion

**Review by geoand — APPROVED** [review]

**Pull request merged**